Example/InstanceID/Tests/FIRInstanceIDTest.m

```diff
@@ -549,6 +549,76 @@ - (void)testDefaultToken_validCachedToken {
   XCTAssertEqualObjects([self.mockInstanceID token], kToken);
 }
 
+/**
+ *  Tests that the callback handler will be invoked when the default token is fetched
+ *  despite the token being unchanged.
+ */
+- (void)testDefaultToken_callbackInvokedForUnchangedToken {
+  XCTestExpectation *defaultTokenExpectation =
+      [self expectationWithDescription:@"Token fetch was successful."];
+
+  __block FIRInstanceIDTokenInfo *cachedTokenInfo = nil;
+
+  [self stubKeyPairStoreToReturnValidKeypair];
+
+  [self mockAuthServiceToAlwaysReturnValidCheckin];
+
+  // Mock Token manager to always succeed the token fetch, and return
+  // a particular cached value.
+
+  // Return a dynamic cachedToken variable whenever the cached is checked.
+  // This uses an invocation-based mock because the |cachedToken| pointer
+  // will change. Normal stubbing will always return the initial pointer,
+  // which in this case is 0x0 (nil).
+  [[[self.mockTokenManager stub] andDo:^(NSInvocation *invocation) {
+    [invocation setReturnValue:&cachedTokenInfo];
+  }] cachedTokenInfoWithAuthorizedEntity:kAuthorizedEntity scope:kFIRInstanceIDDefaultTokenScope];
+
+  [[[self.mockTokenManager stub] andDo:^(NSInvocation *invocation) {
+    self.newTokenCompletion(kToken, nil);
+  }] fetchNewTokenWithAuthorizedEntity:kAuthorizedEntity
+                                 scope:kFIRInstanceIDDefaultTokenScope
+                               keyPair:[OCMArg any]
+                               options:[OCMArg any]
+                               handler:[OCMArg checkWithBlock:^BOOL(id obj) {
+                                 self.newTokenCompletion = obj;
+                                 return obj != nil;
+                               }]];
+
+  __block NSInteger notificationPostCount = 0;
+  __block NSString *notificationToken = nil;
+
+  // Fetch token once to store token state
+  NSString *notificationName = kFIRInstanceIDTokenRefreshNotification;
+  self.tokenRefreshNotificationObserver = [[NSNotificationCenter defaultCenter]
+      addObserverForName:notificationName
```

Comment:
Nit: This may be a bit nicer to express using `[self expectationForNotification:...]`, though probably no need to change since it is used in other tests as well.

Reply:
Keeping it this way for consistency (and laziness...) 

@chliangGoogle FYI